app/components/channel_system_message/channel_system_message.js

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Posts} from 'mattermost-redux/constants';
+import FormattedMarkdownText from 'app/components/formatted_markdown_text';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {ViewTypes} from 'app/constants';
+
+
+export default class MiscSystemMessage extends React.PureComponent {
+
+    render() {
+        const { post, theme, navigator, textStyles } = this.props;
+        console.log('theme:', theme);
+
+        const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+            const stdPadding = 12;
+            return {
+                main: {
+                    flexDirection: 'row',
+                    paddingTop: stdPadding,
+                },
+                iconContainer: {
+                    paddingRight: stdPadding,
+                    paddingLeft: stdPadding,
+                    width: (stdPadding * 2) + ViewTypes.PROFILE_PICTURE_SIZE,
+                },
+                textContainer: {
+                    paddingBottom: 10,
+                    flex: 1,
+                    marginRight: stdPadding,
+                },
+                messageContainer: {
+                    marginTop: 3,
+                },
+                displayName: {
+                    color: theme.centerChannelColor,
+                    fontSize: 15,
+                    fontWeight: '600',
+                },
+                message: {
+                    color: changeOpacity(theme.centerChannelColor, 0.8),
+                    fontSize: 15,
+                    lineHeight: 22,
+                },
+            };
+        });        
+
+        const style = getStyleSheet(theme);
+        console.log('style:', style);
+
+        const renderUsername = (name) => {
```

Comment:
Why make all of these inline functions? Most of them would fit better as methods or as functions outside the component

Reply:
Thank you for the review.
I am implemented most of the changes request. Will do the rest, but wanted to ask a question in the meantime...
Regarding your suggestion above, which I believe is absolutely valid, should I perhaps add a few utility functions (such as `getStyleSheet` and `renderUsername`) outside of components and we could import those as needed? If it makes sense, where would be the best place to stick those?

Reply:
`getStyleSheet` is usually just used by one component, so it's fine as it is. If you'd like to move things like `renderChannelDeletedMessage` into their own files, you could pull `getStyleSheet` into its own file as well, but that's not needed otherwise.

I think it's the same for `renderUsername`. I wouldn't bother moving it to its own file unless it's actually needed in multiple files

Reply:
please take a look

Reply:
Structuring it this way looks good to me